core/src/test/scala/za/co/absa/spline/harvester/conf/YAMLConfigurationTest.scala

```diff
@@ -47,4 +63,31 @@ class YAMLConfigurationTest extends ReadOnlyConfigurationTest {
     givenConf.getKeys("spline").asScala.toSeq shouldNot contain("spline.null_value")
     givenConf.getProperty("spline.null_value") shouldEqual null
   }
+
+  test("get string value as String") {
+    givenConf getString "spline.x.y" shouldEqual "foo"
+    givenConf getString "spline.w.z" shouldEqual "bar"
+  }
+
+  test("get string value as another type") {
+    givenConf getDouble "spline.decimalAsString" shouldEqual 1.2d
+    givenConf getInt "spline.integerAsString" shouldEqual 42
+    givenConf getBoolean "spline.booleanAsString" shouldEqual true
```

Comment:
I think `type` to `type` should be also tested

like:
```
givenConf getDouble "spline.integerAsNumber" shouldEqual 1.2d
```

Reply:
It was too obvious for me to test, but interestingly enough I spotted an error in this one :open_mouth:

Reply:
nah... that was a glitch :laughing:

Reply:
added a unit test